token/program-2022/src/extension/confidential_transfer_fee/processor.rs

```diff
@@ -347,6 +390,64 @@ fn process_harvest_withheld_tokens_to_mint(accounts: &[AccountInfo]) -> ProgramR
     Ok(())
 }
 
+/// Process a [EnableHarvestToMint] instruction.
+fn process_enable_harvest_to_mint(accounts: &[AccountInfo]) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let mint_info = next_account_info(account_info_iter)?;
+    let authority_info = next_account_info(account_info_iter)?;
+
+    check_program_account(mint_info.owner)?;
+    let mint_data = &mut mint_info.data.borrow_mut();
+    let mut mint = StateWithExtensionsMut::<Mint>::unpack(mint_data)?;
+    let confidential_transfer_fee_mint =
+        mint.get_extension_mut::<ConfidentialTransferFeeConfig>()?;
+
+    let maybe_confidential_transfer_fee_authority: Option<Pubkey> =
+        confidential_transfer_fee_mint.authority.into();
+    let confidential_transfer_fee_authority =
+        maybe_confidential_transfer_fee_authority.ok_or(TokenError::NoAuthorityExists)?;
+
+    if !authority_info.is_signer {
+        return Err(ProgramError::MissingRequiredSignature);
+    }
```

Comment:
The comment mentions that this works with multisigs, but this check means that it only works with single authorities. It's fine if we don't want to support multisigs, but we should be clear in the instruction interface